crates/sui-framework/packages/bridge/tests/bridge_tests.move

```diff
@@ -4,373 +4,173 @@
-        init_committee(&mut scenario, @0x0);
-        init_committee(&mut scenario, @0x0); // second time is a no-op
-
-        scenario.end();
+        env.destroy_env();
     }
 
     #[test]
     #[expected_failure(abort_code = bridge::bridge::ENotSystemAddress)]
     fun test_init_committee_non_system_addr() {
-        let mut scenario = test_scenario::begin(@0x0);
-
-        create_bridge_default(&mut scenario);
-        register_committee(&mut scenario);
-        init_committee(&mut scenario, @0xA);
-
+        let mut env = create_env(chain_ids::sui_mainnet(), @0x0);
+        // TODO: this is too brittle, fix it
+        env.setup_validators(
+            vector[
+                create_validator_info(@0xA, 100),
+                create_validator_info(@0xB, 100),
+                create_validator_info(@0xC, 100),
+            ],
+            @0x0,
+        );
+        env.create_bridge(@0x0);
+        env.register_committee();
+        env.init_committee(@0xA);
 
         abort TEST_DONE
     }
 
     #[test]
     #[expected_failure(abort_code = bridge::committee::ECommitteeAlreadyInitiated)]
     fun test_register_committee_after_init() {
-        let mut scenario = test_scenario::begin(@0x0);
-
-        create_bridge_default(&mut scenario);
-        register_committee(&mut scenario);
-        init_committee(&mut scenario, @0x0);
-        register_committee(&mut scenario);
-
+        let mut env = create_env(chain_ids::sui_custom(), @0x0);
+        env.create_bridge_default();
+        env.register_committee();
 
         abort TEST_DONE
     }
 
-    // #[test]
-    // fun test_register_foreign_token() {
-    //     let mut scenario = test_scenario::begin(@0x0);
-
-    //     create_bridge_default(&mut scenario);
-    //     register_committee(&mut scenario);
-    //     init_committee(&mut scenario, @0x0);
-
-    //     scenario.next_tx(@0xAAAA);
-    //     let mut bridge = scenario.take_shared<Bridge>();
-    //     bridge.register_foreign_token<T>(
-    //         tc: TreasuryCap<T>,
-    //         uc: UpgradeCap,
-    //         metadata: &CoinMetadata<T>,
-    //     );
-
-    //     scenario.end();
-    // }
-
-    // #[test]
-    // fun test_execute_send_token() {
-    //     let mut scenario = test_scenario::begin(@0x0);
-
-    //     create_bridge_default(&mut scenario);
-    //     register_committee(&mut scenario);
-    //     init_committee(&mut scenario, @0x0);
-
-    //     scenario.next_tx(@0xAAAA);
-    //     let mut bridge = scenario.take_shared<Bridge>();
-    //     let eth_address = x"0000000000000000000000000000000000000000";
-    //     let btc: Coin<BTC> = coin::mint_for_testing<BTC>(1, scenario.ctx());
-    //     bridge.send_token(
-    //         chain_ids::eth_sepolia(),
-    //         eth_address,
-    //         btc,
-    //         scenario.ctx(),
-    //     );
-    //     test_scenario::return_shared(bridge);
-
-    //     scenario.end();
-    // }
+    #[test]
+    fun test_register_foreign_token() {
```

Comment:
1. do we have a test case for `coin::total_supply(&tc) != 0`?
https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/packages/bridge/sources/treasury.move#L103

2. do we have some test cases for `add_new_token`?

Reply:
1. `test_register_foreign_token_non_zero_supply`
2. `add_default_tokens` to set up the default token does that for the happy path every time you set up the bridge and `test_add_token_price_zero_value`, `test_add_token_malformed_1`, `test_add_token_malformed_2`, `test_add_token_malformed_3` and `test_add_native_token_nop` test the not so happy path